src/cache/mod.rs

```diff
@@ -469,75 +445,6 @@ impl Cache {
         self.guilds.len()
     }
 
-    /// Retrieves a reference to a [`Guild`]'s channel. Unlike [`Self::channel`], this will only
-    /// search guilds for the given channel.
-    ///
-    /// The only advantage of this method is that you can pass in anything that is indirectly a
-    /// [`ChannelId`].
-    ///
-    /// # Examples
-    ///
-    /// Getting a guild's channel via the Id of the message received through a
-    /// [`EventHandler::message`] event dispatch:
-    ///
-    /// ```rust,no_run
-    /// # use serenity::model::prelude::*;
-    /// # use serenity::prelude::*;
-    /// #
-    /// struct Handler;
-    ///
-    /// #[serenity::async_trait]
-    /// impl EventHandler for Handler {
-    ///     async fn message(&self, context: Context, message: Message) {
-    ///         let channel_opt = context.cache.guild_channel(message.channel_id).as_deref().cloned();
-    ///         let channel = match channel_opt {
-    ///             Some(channel) => channel,
-    ///             None => {
-    ///                 let result = message
-    ///                     .channel_id
-    ///                     .say(&context, "Could not find guild's channel data")
-    ///                     .await;
-    ///                 if let Err(why) = result {
-    ///                     println!("Error sending message: {:?}", why);
-    ///                 }
-    ///
-    ///                 return;
-    ///             },
-    ///         };
-    ///     }
-    /// }
-    ///
-    /// # #[cfg(feature = "client")]
-    /// # async fn run() -> Result<(), Box<dyn std::error::Error>> {
-    /// let mut client =
-    ///     Client::builder("token", GatewayIntents::default()).event_handler(Handler).await?;
-    ///
-    /// client.start().await?;
-    /// # Ok(())
-    /// # }
-    /// ```
-    ///
-    /// [`EventHandler::message`]: crate::client::EventHandler::message
```

Comment:
Do we want to keep this example and just move it onto `Cache::channel`?

Reply:
It seems largely useless as an example, as `Message` already has a `to_channel` method, but if you feel strongly about this example I'll move it to `Cache::channel`.